src/merkle.rs

```diff
@@ -0,0 +1,148 @@
+use sha3::{Digest, Sha3_256};
+
+type Hash = Vec<u8>;
+
+struct Merkle {
+    tree: Vec<Hash>, // This will be a binary tree represented as a vector
+}
+
+fn hash(element: &[u8]) -> Hash {
+    Sha3_256::digest(element).to_vec()
+}
+
+// Helper function to hash two internal nodes
+fn hash_internal_node(left: &Hash, right: &Hash) -> Hash {
+    let mut hasher = Sha3_256::new();
+    hasher.update(left);
+    hasher.update(right);
+    hasher.finalize().to_vec()
+}
+
+impl Merkle {
+    /// Creates a new Merkle tree from a list of data elements.
+    /// TODO: Make this function generic to accept any type that can be hashed.
+    ///
+    pub fn new(data: &[Vec<u8>]) -> Self {
+        assert!(data.len() > 1, "Data must have at least 2 elements");
```

Comment:
You can create a merkle tree with only one element, the result is a tree with only one element, that element being the root

Reply:
Done